tests/functional/test_public_apis.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2012-2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import mock
+
+from botocore.session import Session
+from botocore.exceptions import NoCredentialsError
+
+
+class EarlyExit(BaseException):
+    pass
+
+def _test_public_apis_will_not_be_signed(func, kwargs):
+    with mock.patch('botocore.endpoint.Session.send') as _send:
+        _send.side_effect = EarlyExit("we don't care about response here")
+        try:
+            func(**kwargs)
+        except EarlyExit:
+            pass
+        except NoCredentialsError:
+            assert False, "NoCredentialsError should not be triggered"
+        request = _send.call_args[0][0]
```

Comment:
Sorry I ran pep8 and flake8 just now, and they suggest to add one more blank line between each function.  Other than that, there is no warning.  @jamesls Do you see something else that we need to fix?

Meanwhile, I'll work on other comments in this PR. Thanks!